semgrep_output_v1.atd

```diff
@@ -1680,6 +1680,46 @@ type ci_scan_complete_stats = {
     <python repr="dict">
     <ts repr="map">
     option;
+
+  (* since 1.98.0 *)
+  (* In collaboration with the Data Science team, it was suggested
```

Comment:
Oh, ok, I misunderstood the data science piece. Doesn't this only relaly help them if these are in a DB somewhere?

Reply:
Ah! As part of [my project](https://www.notion.so/semgrep/Improved-SSC-Adoption-Metrics-1303009241a8808d84f8feade1a7cd62?pvs=4), the stats in s3 are being piped into snowflake. 

The s3 to snowflake sync is already happening for staging:
![image](https://github.com/user-attachments/assets/05303e48-4a49-41bf-be24-65f3b9e3444c)

The stats json is being transformed into a nice table to derive metrics from!

Reply:
I’m envisioning a future where we no longer store scan stats in our PostgreSQL database. Instead, the stats would be piped directly to S3 and then to Snowflake. This approach would eliminate the need for a JSON column in the scan table. however careful planning is still required, especially since some folks may rely on that column for other purposes.

More context: after discussions with the infra team, we’re increasingly moving towards using Snowflake to centralize all our data. Snowflake’s power make it much better suited for data engineering tasks. For this project, we plan to create several derived tables from the new stats table in Snowflake. 

These tables will be made available in metabase so we can create some dashboards! Essentially, Snowflake will act as our powerful backend, while metabase serves as the frontend, displaying dashboards powered by snowflake’s data.

>  If so, can we make placeholders for the other products as well and follow the same path that Jonas laid out for moving the meta field in ScanRequest to more structured types in project_metadata and scan_metadata.

Placeholders for other products isn't a bad idea. In fact @khoa-semgrep suggested that it would be helpful to eventually divide the stats key into product-specific stats and generic stats applicable across all products

Reply:
It’s worth noting that this is the tentative plan so far (having S3 stats piped to Snowflake). 

The staging test is working well, but if the larger volume of production stats becomes too costly, the backup plan is to revert to using the stats column. We don’t anticipate SSC stats becoming as large as the errors were, but I’d still like to avoid that scenario.

Reply:
> The stats json is being transformed into a nice table to derive metrics from!

Oh, that's very nice! Ok, this seems fine. 

> I’m envisioning a future where we no longer store scan stats in our PostgreSQL database.

FWIW we are depending on several fields that are in the stats field now in the app, so we can't just move it to S3 🙁 In my ideal world, rather than living in JSON, these fields that have simple, size-reasonable data types (bools, ints, floats) would each have their own column in the scan table and only the longer and unbounded-in-size fields would be in S3 (e.g. "errors"). We are already doing that for "errors", but they're being stored to S3 as part of the scan pipeline S3. It may make sense to combine our scan-pipeline buckets (configs, results, stats with errors) with whatever buckets you're using now. The old ones expire after a few weeks, which isn't great. The ones I'm referring to are [here](https://us-west-2.console.aws.amazon.com/s3/buckets/scan-pipeline-prod?region=us-west-2&bucketType=general&tab=objects) and were originally created to support the RFS work.

Reply:
> FWIW we are depending on several fields that are in the stats field now in the app, so we can't just move it to S3 🙁

Ack! I figured some folks might rely on it. 

> these fields that have simple, size-reasonable data types (bools, ints, floats) would each have their own column in the scan table and only the longer and unbounded-in-size fields would be in S3 (e.g. "errors")

that makes sense to me! 

> It may make sense to combine our scan-pipeline buckets (configs, results, stats with errors) with whatever buckets you're using now.

I would love to combine them. At the moment, I went with a [new bucket](https://us-west-2.console.aws.amazon.com/s3/buckets/semgrep-scan-stats-prod?region=us-west-2&bucketType=general&tab=objects) to hold _just_ stats in a very specific format to make ingest easier for snowflake. 

Lots to think about still! @chmccreery appreciate discussion, i'll be sure to capture this in a retro doc for future consideration.